rest-server/src/templates/dockerContainerScript.mustache

```diff
@@ -105,6 +123,73 @@ export PAI_TASK_ROLE_LIST=$PAI_JOB_TASK_ROLE_LIST
 export PAI_CURRENT_CONTAINER_IP=$PAI_CONTAINER_HOST_IP
 export PAI_CURRENT_CONTAINER_PORT=$PAI_CONTAINER_HOST_PORT
 
+function prepare_ssh()
+{
+   mkdir /root/.ssh
+   sed -i 's/PermitRootLogin prohibit-password/PermitRootLogin yes/' /etc/ssh/sshd_config
+   sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
+}
+
+function start_ssh_service()
+{
+  printf "%s %s\n" \
+   "[INFO]" "start ssh service"
+  cat /root/.ssh/$APP_ID.pub >> /root/.ssh/authorized_keys
+  sed -i 's/Port.*/Port '$PAI_CONTAINER_SSH_PORT'/' /etc/ssh/sshd_config
+  echo "sshd:ALL" >> /etc/hosts.allow
+  service ssh restart
+}
+
+function hdfs_upload_atomically()
+{
+   printf "%s %s\n%s %s\n%s %s\n" \
+    "[INFO]" "upload ssh key to hdfs" \
+    "[INFO]" "destination path is ${2}" \
+    "[INFO]" "source path is ${1}"
+   tempFolder=${2}"_temp"
+   if hdfs dfs -test -d $tempFolder ; then
+     printf "%s %s\n" \
+    "[WARNING]" "$tempFolder already exists, overwriting..."
+     hdfs dfs -rm -r $tempFolder || exit 1
+   fi
+   hdfs dfs -put ${1} $tempFolder || exit 1
+   hdfs dfs -mv $tempFolder ${2} || exit 1
+}
+
+# Start sshd in docker container
+prepare_ssh
+hdfs_ssh_folder=${HDFS_LAUNCHER_PREFIX}/${PAI_USER_NAME}/${PAI_JOB_NAME}/ssh/${APP_ID}
+printf "%s %s\n%s %s\n%s %s\n" \
+  "[INFO]" "hdfs_ssh_folder is ${hdfs_ssh_folder}" \
+  "[INFO]" "task_role_no is ${task_role_no}" \
+  "[INFO]" "PAI_TASK_INDEX is ${PAI_TASK_INDEX}"
+# let taskRoleNumber=0 and taskindex=0 execute upload ssh files
+if [ ${task_role_no} -eq 0 ] && [ ${PAI_TASK_INDEX} -eq 0 ]; then
+  printf "%s %s %s\n%s\n" \
+    "[INFO]" "task_role_no:${task_role_no}" "PAI_TASK_INDEX:${PAI_TASK_INDEX}" \
+    "Execute upload key pair ..."
+  ssh-keygen -N '' -t rsa -f ~/.ssh/$APP_ID
+  hdfs dfs -mkdir -p "${hdfs_ssh_folder}" || exit 1
+  hdfs_upload_atomically "/root/.ssh/" "${hdfs_ssh_folder}/.ssh"
+else
+  # waiting for ssh key-pair ready
+  while ! hdfs dfs -test -d ${hdfs_ssh_folder}/.ssh ; do
+    echo "[INFO] waitting for ssh key ready"
+    sleep 10
+  done
+  printf "%s %s\n%s %s\n" \
+      "[INFO]" "ssh key pair ready ..." \
+      "[INFO]" "begin to download ssh key pair from hdfs ..."
+  hdfs dfs -get "${hdfs_ssh_folder}/.ssh/" "/root/" || exit 1
+fi
+# Start ssh service
+start_ssh_service
+# Generate ssh connect info file in "PAI_CONTAINER_ID-PAI_CURRENT_CONTAINER_IP-PAI_CONTAINER_SSH_PORT" format on hdfs
+hdfs dfs -touchz ${hdfs_ssh_folder}/$PAI_CONTAINER_ID-$PAI_CONTAINER_HOST_IP-$PAI_CONTAINER_SSH_PORT || exit 1
+
+# Write env to bashrc
+env | sed "s/^/export /" >> ~/.bashrc
```

Comment:
`grep "^PAI"` before `sed`?

Reply:
I discussed with Bin, if we only keep environment started with "PAI", some configurations like hdfs will be incorrect when SSH connect into container.

Reply:
ok